### PR TITLE
Add debug to worker timeout and startup. Closes #2528

### LINKED ISSUE #2528: Improve log messages to help debug worker timeouts
**Is your feature request related to a problem? Please describe.**

Users sometimes encounter `Terminating timed out worker` messages in puma logs. These messages are generated in one of two cases:

1) an already-booted worker failed to report itself to the master process within the last `worker_timeout` seconds (default 60)
2) a new worker failed to boot within `worker_boot_timeout` seconds (default 60).

If a user is getting these messages in the log, it's not really clear which situation they ran into and makes it harder to diagnose the problem. See #2527 for an example

**Describe the solution you'd like**

1. Log messages should clearly indicate if a worker boot timed out or if a worker failed to "check-in" after boot

Example:

* `Terminating timed out worker (worker failed to check in within 60 seconds): 2006`
* `Terminating timed out worker (worker failed to boot within 60 seconds): 2008`

2. The other thing that would be nice is just writing to the logs the amount of time that it look to boot a worker (when it was successful)

* `Worker 0 (pid: 20934) booted in 12.34 seconds, phase: 2`

This would be useful to help operators know if their workers are booting close to the timeout threshold or not

### PR DESCRIPTION
**Pull request by ylecuyer**
### Description
Fixes #2528 adding the reason for worker timeout and the startup time when worker boots

### Your checklist for this pull request

- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` or `[ci skip]` to the pull request title.
- [x] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [x] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: test/helpers/integration.rb [hunk 1]
```diff
@@ -46,6 +46,10 @@ def teardown
 
   private
 
+  def silent_and_checked_system_command(*args)
+    assert(system(*args, out: File::NULL, err: File::NULL))
```
Comment (ylecuyer) [inline test/helpers/integration.rb:50]: Removed err as bundler keeps on complaining my version is not the same as the one in the `Gemfile.lock`
I also added an assert to make sure the command exited with success

File: History.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 * Bugfixes
   * Add `#flush` and `#sync` methods to `Puma::NullIO`  ([#2553])
+* Minor
```
Comment (dentarg) [inline History.md:5]:
```suggestion
* Features
```

**Review by dentarg — APPROVED** [review]
LGTM

**Review by dentarg — CHANGES_REQUESTED** [review]
Looks like failing tests needs to be addressed

**Comment by ylecuyer** [comment]
> Looks like failing tests needs to be addressed

I have fixed the ruby < 2.4 versions. Sadly I can't test locally as I don't have the right openssl version to compile the old rubies. Let's see if the CI is happy this time ;)

**Comment by MSP-Greg** [comment]
CI is fine, current CI has intermittent failures/errors for all platforms/versions.  See:

https://github.com/MSP-Greg/puma/actions/runs/626438062, failures here passed, ubuntu-18.04 2.6 failed.

@cjlarose - given #2528, if you have time to review...

File: lib/puma/cluster.rb [hunk 3]
```diff
@@ -434,7 +434,7 @@ def run
                 case req
                 when "b"
                   w.boot!
-                  log "- Worker #{w.index} (PID: #{pid}) booted, phase: #{w.phase}"
+                  log "- Worker #{w.index} (PID: #{pid}) booted in #{w.uptime.truncate(*(RUBY_VERSION >= "2.4." ? [2] : []))}s, phase: #{w.phase}"
```
Comment (dentarg) [inline lib/puma/cluster.rb:437]:
```suggestion
                  log "- Worker #{w.index} (PID: #{pid}) booted in #{w.uptime.round(2)}s, phase: #{w.phase}"
```

Should also get us decimals when using Ruby < 2.4

**Comment by dentarg** [comment]
> Sadly I can't test locally as I don't have the right openssl version to compile the old rubies

If you can use Docker, it is as simple as `docker run --rm -it ruby:2.2` (it will start `irb` as default)

**Comment by cjlarose** [comment]
> @cjlarose - given #2528, if you have time to review...

Oh man I forgot I wrote that issue. I'll do some local testing on this, but on the surface, this looks great!

**Review by cjlarose — APPROVED** [review]
Awesome work. Thanks for doing this!

**Review by dentarg — APPROVED** [review]
I assume the CI failures are not related

(18.04 2.7 flaky `test_hot_restart_does_not_drop_connections` and Win 2019 2.3 `SignalException: SIGSEGV`, flake too?)

**Pull request merged**